src/ape/contracts/base.py

```diff
@@ -628,6 +628,28 @@ def poll_logs(
             yield from self.range(new_block.number, stop=new_block.number + 1)
 
 
+class CustomError:
```

Comment:
Attempting to make this an instance of `ContractLogicError` and have that be what actually gets raised, but the issue there is that this does not know the inputs - only the input types (abi). This class is more like a `CustomErrorType` class rather than an actual exception class.

Reply:
actually a good name for the class